eosdis-nasa/earthdata-pub-dashboard
# EDPUB-1282: Reimplement custom MFA support

# Description

Reimplemented custom MFA support to help resolve lost device MFA use case which is not supported by AWS native MFA support. This PR has an accompanying PR in the API.

## Spec

See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1283

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Push this branch as well as the accompanying API branch to SIT
3. Test signing into your MFA enabled account.
4. Verify that your sign in is redirected as expected.
5. Sign out of your MFA enabled account.
6. Update your AWS Cognito user's MFA configuration to inactive.
7. Test signing into your account.
8. Verify that upon signing in, you are redirected to the MFA setup page.
9. Verify that registering a new MFA device works as expected.

**NOTE: Your previous MFA setup will no longer work.**

---
